rotary/repair.py

```diff
@@ -780,29 +823,29 @@ def main():
     parser = parse_cli()
     args = parser.parse_args()
 
-    # Set user variables
-    long_read_filepath = args.long_read_filepath
-    assembly_fasta_filepath = args.assembly_fasta_filepath
-    assembly_info_filepath = args.assembly_info_filepath
-    assembly_info_type = 'custom' if args.custom_assembly_info_file is True else 'flye'
-    output_dir = args.output_dir
-    length_thresholds = [int(x) for x in args.length_thresholds.split(',')]
-    keep_failed_contigs = args.keep_going_with_failed_contigs
-    cli_tool_settings_dict = {'flye_read_mode': args.flye_read_mode,
-                              'flye_read_error': args.flye_read_error,
-                              'circlator_min_id': args.circlator_min_id,
-                              'circlator_min_length': args.circlator_min_length,
-                              'circlator_ref_end': args.circlator_ref_end,
-                              'circlator_reassemble_end': args.circlator_reassemble_end}
-    threads = args.threads
-    threads_mem_mb = int(args.threads_mem * 1024)
-    verbose = args.verbose
-
     # Startup checks
-    if verbose is True:
-        logger.setLevel(logging.DEBUG)
+    if args.verbose is True:
+        stream_handler.setLevel(logging.DEBUG)
     else:
-        logger.setLevel(logging.INFO)
+        stream_handler.setLevel(logging.INFO)
+
+    # Check output dir
+    output_dir_exists = os.path.isdir(args.output_dir)
+
+    if (output_dir_exists is True) & (args.overwrite is False):
+
+        logger.error(f'Output directory already exists: "{args.output_dir}". Will not continue. Set the '
+                     f'--overwrite flag at your own risk if you want to use an existing directory.')
+
+        raise FileExistsError
+
+    os.makedirs(args.output_dir, exist_ok=True)
+
+    # Start log file in the output dir
+    file_handler = logging.FileHandler(filename=os.path.join(args.output_dir, 'repaired.log'), mode='w')
+    file_handler.setFormatter(formatter)
+    file_handler.setLevel(logging.DEBUG)
```

Comment:
As discussed in other comments, one wants to write the info-level logs, not debug logs, by default. This way, your log files don't get massive by printing everything. So your ```--verbose``` flag should control this log level to.

Reply:
See conversation about line 25 above